dmutils/api_stubs.py

```diff
@@ -209,11 +209,19 @@ def brief(status="draft",
             "clarificationQuestions": clarification_questions or [],
         }
     }
+
     if status in ("live", "closed"):
-        brief['briefs']['publishedAt'] = "2016-03-29T10:11:14.000000Z"
-        brief['briefs']['applicationsClosedAt'] = "2016-04-07T00:00:00.000000Z"
-        brief['briefs']['clarificationQuestionsClosedAt'] = "2016-04-02T00:00:00.000000Z"
-        brief['briefs']['clarificationQuestionsAreClosed'] = clarification_questions_closed
+        brief["briefs"]["publishedAt"] = "2016-03-29T10:11:14.000000Z"
+        brief["briefs"]["applicationsClosedAt"] = "2016-04-07T00:00:00.000000Z"
+        brief["briefs"]["clarificationQuestionsClosedAt"] = "2016-04-02T00:00:00.000000Z"
+        brief["briefs"]["clarificationQuestionsAreClosed"] = clarification_questions_closed
+        brief["briefs"]["clarificationQuestionsPublishedBy"] = "2016-04-02T00:00:00.000000Z"
```

Comment:
Sorry, I wasn't clear about what each status needs (it's certainly not clear from the API!) Each of `live`, `closed`, `awarded`, `unsuccessful`, `withdrawn` and `cancelled` will have these 5 fields in the serialization, because they will all have the `published_at` datestamp set. It's only `draft` that doesn't have that published date.

The `withdrawnAt`/`unsuccessfulAt`/`cancelledAt` dates below are fine. `awarded` briefs have a `awardedBriefResponseId` which is probably not a good idea to add in to an api stub, the test using the stub can deal with that itself.

Reply:
How's it look now?